app/src/main/java/com/hedvig/app/feature/embark/EmbarkViewModel.kt

```diff
@@ -260,11 +264,15 @@ abstract class EmbarkViewModel(
     }
 
     private fun createOfferEvent(): Event {
-        val quoteCartId = valueStore.get(QUOTE_CART_ID_KEY)
-        return Event.Offer(
-            quoteCartId = quoteCartId?.let { QuoteCartId(it) },
-            selectedContractTypes = emptyList(),
-        )
+        val quoteCartId = valueStore.get(QUOTE_CART_ID_KEY)?.let(::QuoteCartId)
+        return if (quoteCartId != null) {
+            Event.Offer(
+                quoteCartId = quoteCartId,
+                selectedContractTypes = emptyList(),
+            )
+        } else {
+            Event.Error()
+        }
```

Comment:
Again here, if we've reached the point that we call `createOfferEvent` and we don't have a QuoteCartId it seems like quite a bigger problem. Since we're using only that for now, we can't have an offer without an ID. So safe to error here as well afaik.